src/Cli/dotnet/PerformanceLogEventSource.cs

```diff
@@ -0,0 +1,341 @@
+                }
+            }
+        }
+
+        [Event(26)]
+        internal void MemoryConfiguration(int memoryLoad, int availablePhysicalMB, int totalPhysicalMB)
+        {
+            WriteEvent(26, memoryLoad, availablePhysicalMB, totalPhysicalMB);
+        }
+    }
+
+    internal class PerformanceLogStartupInformation
+    {
+        public PerformanceLogStartupInformation(DateTime mainTimeStamp)
+        {
+            // Save the main timestamp.
+            MainTimeStamp = mainTimeStamp;
+
+            // Attempt to load an assembly.
+            // Ideally, we've picked one that we'll already need, so we're not adding additional overhead.
+            MeasureModuleLoad();
+        }
+
+        internal DateTime MainTimeStamp { get; private set; }
+        internal Assembly TimedAssembly { get; private set; }
+        internal TimeSpan AssemblyLoadTime { get; private set; }
+
+        private void MeasureModuleLoad()
+        {
+            // Make sure the assembly hasn't been loaded yet.
+            string assemblyName = "Microsoft.DotNet.Configurer";
+            try
+            {
+                foreach (Assembly loadedAssembly in AppDomain.CurrentDomain.GetAssemblies())
+                {
+                    if (loadedAssembly.GetName().Name.Equals(assemblyName))
+                    {
+                        // If the assembly is already loaded, then bail.
+                        return;
+                    }
+                }
+            }
+            catch
+            {
+                // If we fail to enumerate, just bail.
+                return;
+            }
+
+            Stopwatch stopWatch = Stopwatch.StartNew();
+            Assembly assembly = null;
+            try
+            {
+                assembly = Assembly.Load(assemblyName);
+            }
+            catch
+            {
+                return;
+            }
+            stopWatch.Stop();
+            if (assembly != null)
+            {
+                // Save the results.
+                TimedAssembly = assembly;
+                AssemblyLoadTime = stopWatch.Elapsed;
+            }
+        }
+    }
+
+    internal static class Interop
+    {
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct MEMORYSTATUSEX
+        {
+            // The length field must be set to the size of this data structure.
+            internal uint dwLength;
+            internal uint dwMemoryLoad;
+            internal ulong ullTotalPhys;
+            internal ulong ullAvailPhys;
+            internal ulong ullTotalPageFile;
+            internal ulong ullAvailPageFile;
+            internal ulong ullTotalVirtual;
+            internal ulong ullAvailVirtual;
+            internal ulong ullAvailExtendedVirtual;
+        }
+
+        [DllImport("kernel32.dll")]
+        internal static extern bool GlobalMemoryStatusEx(ref MEMORYSTATUSEX lpBuffer);
```

Comment:
Test this on windows nano

Reply:
Yes, that's a great idea.

---
In reply to: [505710841](https://github.com/wli3/sdk/pull/1#discussion_r505710841) [](ancestors = 505710841)

Reply:
Have you tested it?